staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -330,8 +333,8 @@ func (config *DirectClientConfig) Tenant() (string, bool, error) {
 		return "", false, err
 	}
 
-	if len(configContext.Tenant) == 0 {
-		return metav1.TenantSystem, false, nil
+	if configContext.Tenant == metav1.TenantAllExplicit {
+		return "", false, fmt.Errorf("%q is not a valid tenant name", metav1.TenantAllExplicit)
```

Comment:
nit: %q can be replaced with metav1.TenantAllExplicit as it's a constant.

Reply:
It is better to keep it like this. If someday in the future we change to TenantAllExplicit = "somethging-else", we don't need the code here.